library/defrag.c

```diff
@@ -193,10 +193,11 @@ int eblob_defrag(struct eblob_backend *b)
 			continue;
 		}
 
-		/* skips sorted bases if defrag for them is not needed. Always defrag unsorted bases and
+		/* skips sorted bases if defrag for them is not needed. Defrag unsorted bases and
 		 * bases that could be merged or defraged.
 		 **/
-		if (want == EBLOB_DEFRAG_NOT_NEEDED && datasort_base_is_sorted(bctl) == 1)
+		if (want == EBLOB_DEFRAG_NOT_NEEDED &&
+			(b->want_defrag == EBLOB_DEFRAG_STATE_DATA_COMPACT || datasort_base_is_sorted(bctl) == 1))
```

Comment:
You should write detailed description of this **very obscure** flag where it is defined and where it is used.
If I understood correctly, it prevents blob defragmentation/sorting when client requests blobs to be sorted, but given blob albeit not sorted does not require defragmentation.